shared/fs/filepreview/index.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../../styles'
+import {Box, Button, Icon, Text} from '../../common-adapters'
+import {formatTimeForMessages} from '../../util/timestamp'
+import FolderHeader from '../header/header-container'
+
+type FilePreviewHeaderProps = {
+  title: string,
+  desc: string,
+}
+
+const FilePreviewHeader = ({title, desc}: FilePreviewHeaderProps) => (
+  <Box>
+    <Box style={filePreviewHeaderStyle}>
+      <Text type="BodyBig">{title}</Text>
+      <Text type="BodySmall">{desc}</Text>
+    </Box>
+  </Box>
+)
+
+type FilePreviewProps = {
+  path: Types.Path,
+  meta: Types.PathItemMetadata,
+}
+
+const humanReadableFileSize = meta => {
+  const kib = 1024
+  const mib = kib * kib
+  const gib = mib * kib
+  const tib = gib * kib
+
+  if (!meta) return ''
+  const size = meta.size
+  if (size >= tib) return `${Math.round(size / tib)}tb`
+  if (size >= gib) return `${Math.round(size / gib)}gb`
+  if (size >= mib) return `${Math.round(size / mib)}mb`
+  if (size >= kib) return `${Math.round(size / kib)}kb`
+  return '' + size
+}
+
+class FilePreview extends React.PureComponent<FilePreviewProps> {
+  render() {
+    const {path, meta} = this.props
+    const fileName = Types.getPathName(path)
+    let desc = 'loading'
+    if (meta) {
+      desc = 'Modified on ' + formatTimeForMessages(meta.lastModifiedTimestamp)
```

Comment:
I'm realizing we almost steped on each other's foot. I added a `formatTimeForFS` in [my PR](https://github.com/keybase/client/pull/10845/files#diff-53fb679b63d4b7f0211fa360155ce50dR48) that uses `moment`. The format also seems to be a bit different. No big deal though. It's a simple change if we need it in the future.

Reply:
ok.